src/Console/Application.php

```diff
@@ -34,13 +40,60 @@ class Application extends BaseApplication
      */
     private $container;
 
+    /**
+     * @var SymfonyStyle
+     */
+    private $io;
+
+    /**
+     * @var bool
+     */
+    private $isXdebugLoaded;
+
+    /**
+     * @var bool
+     */
+    private $isDebuggerDisabled;
+
     public function __construct(Container $container, string $name = self::NAME, string $version = self::VERSION)
     {
         $this->container = $container;
+        $this->isDebuggerDisabled = empty((string) getenv(XdebugHandler::ENV_DISABLE_XDEBUG));
 
         parent::__construct($name, $version);
     }
 
+    public function run(InputInterface $input = null, OutputInterface $output = null)
+    {
+        if (null === $input) {
+            $input = new ArgvInput();
+        }
+
+        if (null === $output) {
+            $output = new ConsoleOutput();
+        }
+
+        $this->io = new SymfonyStyle($input, $output);
+
+        if (PHP_SAPI === 'phpdbg') {
```

Comment:
Could you please explain why do we need to show these messages? It looks like a warning, but it's not clear what to do with it for the final user.

Reply:
it's implementation of the @theofidry's third proposal.

```
3. Which was actually the reason why I wanted to open this issue: as when xdebug is enabled we disable it later on for perf improvements, WDYT of enabling it if neither phpdbg is found and xdebug enabled instead of the error message above? We could then add an information "running with xdebug blabla" when running infection. If the user is not happy with it he can always cancel it and restart with phpdbg.
```

Reply:
I updated the comment with the following

> As I understand from Theo's comment, it needs to be displayed when user has it disabled, but infection enables it automatically

Reply:
@theofidry seems like we have a misunderstanding here. What did you mean ? 😆

Reply:
I got it like this: "if script is run without phpdbg and xdebug is *not* enabled - enable it automatically and show the message that it was enabled by Infection"

Reply:
if so my comment from the issue is fully described why we can't do this and we can remove all these lines

Reply:
@theofidry ?

Reply:
Right, we cannot enable xdebug or phpdbg for the user so instead we're sticking with bailing out with a pretty message to the user.

So my comment was:
- Adding a note to know which of xdebug or phpdbg is used
- Add a friendly note when bailing out if neither xdebug and phpdbg are used

Both are already done, @sidz made it more pretty if I understood that right.

Reply:
Seems like I miss the point. It is still not clear for me why this orange (like warning) message is displayed and what value does it bring.

`! [NOTE] You are running Infection with phpdbg enabled.`

* If I have xdebug enabled globally - it's obvious that infection is run with Xdebug as well. 
* If I run `phpdbg -qrrr bin/infection ...` - it's obvious that it is run with phpdbg enabled.

-------

When you run let's say PHPCSFixer with Xdebug, it says "Xdebug has a major performance impact..." and it's clear that user should disable it. But what the use should do with the message above?

Reply:
Then maybe it should be displayed as plain text to not be interpreted as a warning. The idea is just to write this info somewhere.

It's not always obvious what Infection is running with especially if it's hidden behind an alias. For example in case of box running Infection is done with `make infection`, you don't pick xdebug or phpdbg neither do I know if xdebug is enabled or not (I have to enable/disable it very often so it's not always clear).

Also for the error message, maybe we should point to the doc or replicate with the doc how to run it with phpdbg:

```
[ERROR] Neither phpdbg or xdebug has been found. One of those is required by Infection in order to generate coverage data. Either:
- Enable xdebug and run infection again
- Use phpdbg: phpdbg -qrr infection [...]
```

Reply:
Got it. My vote is for changing the colour to something more positive ;)

> Also for the error message, ...

👍

Reply:
done.

p.s. maybe we should show these notes only under verbose mode? thoughts?
p.p.s. You know I start think that this message is really useless :)

Reply:
> p.s. maybe we should show these notes only under verbose mode

like it